torch/utils/data/datapipes/iter/selecting.py

```diff
@@ -12,28 +12,54 @@ class FilterIterDataPipe(MapIterDataPipe):
 
     Iterable DataPipe to filter elements from datapipe according to filter_fn.
     args:
-        datapipe: Iterable DataPipe being filterd
+        datapipe: Iterable DataPipe being filtered
         filter_fn: Customized function mapping an element to a boolean.
         fn_args: Positional arguments for `filter_fn`
         fn_kwargs: Keyword arguments for `filter_fn`
+        drop_empty_batches: By default, drops batch if it is empty after filtering instead of keeping an empty list
+        nesting_level: Determines which level the fn gets applied to, by default it applies to the top level (= 0)
```

Comment:
So what does `-1` do? You seem to also accept that in the body of this function.

Reply:
It applies the `filter` on the lowest nesting level. I can add a note in the comment.

Reply:
Does, e.g., -2 work?

Reply:
No, in this implementation only -1 works